### PR TITLE
809 remove strategy with no credentials

### LINKED ISSUE #809: User creation without credentials
I've seen a weird  behaviour, I created an user with admin profile, no kuid provided nor credentials, I try to do that a second time but a toast appears and said that I must add an username, same toast when i tried to update the first created account (without credentials)

### PR DESCRIPTION
**Pull request by berthieresteban**
## What does this PR do ?
fix #809 

Before: Create an user with blank username/password for local strategy -> create an user with empty string as credentials
Now: Create an user with blank username/password for local strategy -> just remove strategy from credentials during user creation

UI/UX of that card should be adapted -> #874 

### How should this be manually tested?
  - Step 1 : Go to security/users
  - Step 2 : play with the strategy credentials during user creation
  - Step 3 : Should be able to create user without credentials

### Other changes
display kuid when editing user

### CODE REVIEW

**Review by rolljee — APPROVED** [review]

**Comment by berthieresteban** [comment]
Draft -> I forgot to write some tests

**Review by alexandrebouthinon — APPROVED** [review]

File: src/components/Security/Users/CreateOrUpdate.vue [hunk 1]
```diff
@@ -185,7 +185,12 @@ export default {
       this.$v.kuid.$model = value
     },
     onCredentialsChanged(payload) {
-      this.credentials[payload.strategy] = { ...payload.credentials }
+      const notEmptyFields = Object.keys(payload.credentials).filter(c => payload.credentials[c] !== "").length > 0
```
Comment (xbill82) [inline src/components/Security/Users/CreateOrUpdate.vue:188]: I'd rather do this check right before submitting, don't you agree?

**Review by xbill82 — CHANGES_REQUESTED** [review]

File: src/components/Security/Users/CreateOrUpdate.vue (hunk 1 above)
Comment (Leodau) [reply src/components/Security/Users/CreateOrUpdate.vue:188]: I think he was trying to centralize the logic.

**Review by rolljee — APPROVED** [review]

File: src/components/Security/Users/CreateOrUpdate.vue (hunk 1 above)
Comment (berthieresteban) [reply src/components/Security/Users/CreateOrUpdate.vue:188]: Yes @Leodau
I can replace that on submit

Comment (xbill82) [reply src/components/Security/Users/CreateOrUpdate.vue:188]: 🤔 But... `onCredentialsChanged` is called on submit?

Comment (berthieresteban) [reply src/components/Security/Users/CreateOrUpdate.vue]: Nope, `onCredentialsChanged` is called every time credentials input changed
I replaced it on submit

**Review by Njuelle — APPROVED** [review]

**Review by xbill82 — APPROVED** [review]

**Pull request merged**